allendomingo/StoreInventory
# [NET-0008] Add authentication and permissions

## Description
Resolves #8 

**Changelog:**
- [x] Implemented signup and login endpoints.
- [x] Utility functions should be implemented, which do role verification, and check whether a user is logged in.
- [x] GET, POST, and DELETE endpoints should be implemented for /users. Functionalities for each involve basic Read, Create, and Delete functions respectively.
- [x] GET, PUT, and DELETE endpoints should be implemented for /users/:userId. Functionalities for each involve basic Read, Update, and Delete functions respectively, for specific users identified with userId.
- [x] Implement the ff. roles: user, operator, manager, and admin

**How to Test:**
1. See swagger for schemas
2. On postman, create a user using /users/signup endpoints
3. Login using /users/login
4. Place the token in Authorization -> Token for succeeding API calls
